canister/src/unstable_blocks.rs

```diff
@@ -71,6 +71,19 @@ impl UnstableBlocks {
         self.network
     }
 
+    /// Returns all blocks in the tree with the number of confirmations
+    /// separated by heights.
+    pub fn blocks_with_confirmations_by_height(
+        &self,
+        track_until_height: usize,
+    ) -> Vec<Vec<(&Block, u32)>> {
+        let mut blocks_with_confirmations_by_height: Vec<Vec<(&Block, u32)>> =
+            vec![vec![]; track_until_height];
+        self.tree
+            .blocks_with_depth_by_height(&mut blocks_with_confirmations_by_height, 0);
+        blocks_with_confirmations_by_height
```

Comment:
I'd move this logic into the `BlockTree`, as that is code required to use `blocks_with_depth_by_height`. Adding unit tests to it would also be nice :)

Reply:
You are right, will do it.